core/activity/src/provider/service.rs

```diff
@@ -74,7 +78,62 @@ pub fn bind_gsb(db: &DbExecutor, tracker: TrackerRef) {
     counter!("activity.provider.destroyed.by_requestor", 0);
     counter!("activity.provider.destroyed.unresponsive", 0);
 
-    local::bind_gsb(db, tracker);
+    local::bind_gsb(&db.clone(), tracker.clone());
+
+    let db = db.clone();
+    let tracker = tracker.clone();
+    tokio::task::spawn_local(initial_checkup(db, tracker));
+}
+
+async fn initial_checkup(db: DbExecutor, tracker: TrackerRef) {
+    log::info!("Initial activities checkup");
+    let agreements_res = get_approved_agreements().await;
+    let Ok(agreements) = agreements_res else {
+        log::error!("Acitivties initial checkup failure. Err: {agreements_res:?}");
+        return;
+    };
+
+    for agreement in agreements {
+        if let Err(err) = monitor_alive_activities(agreement, db.clone(), tracker.clone()).await {
+            log::error!("Acitivties initial checkup error. Err: {err}")
+        };
+    }
+}
+
+async fn get_approved_agreements() -> anyhow::Result<Vec<AgreementListEntry>> {
+    let state = Some(AgreementState::Approved);
+    let list_agreements_query = market::ListAgreements {
+        state,
+        ..Default::default()
+    };
+    let agreements = bus::service(market::BUS_ID)
+        .send(list_agreements_query)
+        .await??;
+    Ok(agreements)
+}
+
+async fn monitor_alive_activities(
+    agreement: AgreementListEntry,
+    db: DbExecutor,
+    tracker: TrackerRef,
+) -> anyhow::Result<()> {
+    log::info!("Initial activities checkup for agreement {}", agreement.id);
+    let activities_ids = get_activities_for_agreement(&db, &agreement.id).await?;
+    let activity_state_dao = db.as_dao::<ActivityStateDao>();
+    for activity_id in activities_ids {
+        let activity_state = activity_state_dao.get(&activity_id).await?;
+        if activity_state.alive() {
+            log::info!("Initial activty {activity_id} monitoring");
+            tokio::task::spawn_local(monitor_activity(
+                db.clone(),
+                tracker.clone(),
+                activity_id,
+                agreement.provider_id,
```

Comment:
It is the reason I added `provider_id` (and `requestor_id`) to `AgreementListEntry`.
https://github.com/golemfactory/ya-client/pull/171/files
Without doing it I would need to do `ListAgreements`, and then for every agreement do `GetAgreement` call only to get provider_id.
@nieznanysprawiciel is it ok to change `ya-client`?

Reply:
No.